packages/flutter/lib/src/material/action_buttons.dart

```diff
@@ -0,0 +1,427 @@
+    if (iconBuilder != null) {
+      return iconBuilder(context);
+    }
+
+    final IconData data = getIcon(context);
+    final String? semanticsLabel;
+    // This can't use the platform from Theme because it is the Android OS that
+    // expects the duplicated tooltip and label.
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+        semanticsLabel =
+            getAndroidSemanticsLabel(MaterialLocalizations.of(context));
+        break;
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+        semanticsLabel = null;
+        break;
+    }
+
+    return Icon(data, semanticLabel: semanticsLabel);
+  }
+}
+
+/// A "back" icon that's appropriate for the current [TargetPlatform].
+///
+/// The current platform is determined by querying for the ambient [Theme].
+///
+/// See also:
+///
+///  * [BackButton], an [IconButton] with a [BackButtonIcon] that calls
+///    [Navigator.maybePop] to return to the previous route.
+///  * [IconButton], which is a more general widget for creating buttons
+///    with icons.
+///  * [Icon], a Material Design icon.
+///  * [ThemeData.platform], which specifies the current platform.
+class BackButtonIcon extends StatelessWidget {
+  /// Creates an icon that shows the appropriate "back" image for
+  /// the current platform (as obtained from the [Theme]).
+  const BackButtonIcon({ super.key });
+
+  @override
+  Widget build(BuildContext context) {
+    return _ActionIcon(
+      iconBuilderCallback: (ActionIconThemeData? actionIconTheme) {
+        return actionIconTheme?.backButtonIconBuilder;
+      },
+      getIcon: (BuildContext context) {
+        if (kIsWeb) {
+          // Always use 'Icons.arrow_back' as a back_button icon in web.
+          return Icons.arrow_back;
+        }
+        switch (Theme.of(context).platform) {
+          case TargetPlatform.android:
+          case TargetPlatform.fuchsia:
+          case TargetPlatform.linux:
+          case TargetPlatform.windows:
+            return Icons.arrow_back;
+          case TargetPlatform.iOS:
+          case TargetPlatform.macOS:
+            return Icons.arrow_back_ios;
+        }
+      },
+      getAndroidSemanticsLabel: (MaterialLocalizations materialLocalization) {
+        return materialLocalization.backButtonTooltip;
+      },
+    );
+  }
+}
+
+/// A Material Design back icon button.
+///
+/// A [BackButton] is an [IconButton] with a "back" icon appropriate for the
+/// current [TargetPlatform]. When pressed, the back button calls
+/// [Navigator.maybePop] to return to the previous route unless a custom
+/// [onPressed] callback is provided.
+///
+/// The [onPressed] callback can, for instance, be used to pop the platform's navigation stack
+/// via [SystemNavigator] instead of Flutter's [Navigator] in add-to-app
+/// situations.
+///
```

Comment:
This would be a good place to explain that the style's iconColor parameter overrides this class's color parameter for states where the foregroundColor resolves to non-null.

A test for this case (where both color and ButtonStyle.iconColor are specified) should be included as well.